open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java

```diff
@@ -159,14 +161,19 @@ private void initializeTopic()
                  * The consumer group defines which list of events that this connector is processing.  A particular server
                  * wants to keep reading from the same list.  Thus it needs to be passed the group.id it used
                  * the last time it ran.  This is supplied in the connection object as the serverIdProperty.
+                 *
+                 * If explicitly set using the standard kafka property, then use that.  Otherwise, use our default.
                  */
-                serverId = (String) configurationProperties.get(KafkaOpenMetadataTopicProvider.serverIdPropertyName);
-                consumerProperties.put("group.id", serverId);
+
+                if (StringUtils.isEmpty((String)consumerProperties.get("group.id"))) {
+                    serverId = (String) configurationProperties.get(KafkaOpenMetadataTopicProvider.serverIdPropertyName);
```

Comment:
I think this line must be outside of the `if`-block. Otherwise the `serverId` is never set.

Reply:
Agreed! It gets used later.